arkouda-docker/build_docker_image.py

```diff
@@ -0,0 +1,196 @@
+        result = subprocess.run(args=['docker','build',
+                                               '--build-arg', f'CHPL_BASE_IMAGE=ubuntu:22.04',
+                                               '--build-arg', f'CHPL_VERSION={chapelVersion}',
+                                               '--build-arg', f'CHPL_UDP_IMAGE_REPO={dockerRepo}',
+                                               '-f',file,
+                                               '-t', f'{dockerRepo}/{file}:{chapelVersion}', '.'], stdout=subprocess.DEVNULL)
+        print(result)
+
+
+    dockerTag = generateBuildTag(dockerRepo=dockerRepo, file=file, tag=tag,distro=distro)
+
+    if file == 'arkouda-full-stack':   
+        buildArkoudaFullStack(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag,distro=distro,tag=tag)
+    elif file == 'arkouda-smp-server':
+        buildArkoudaSmpServer(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag,distro=distro,tag=tag)
+    elif file == 'arkouda-udp-server':
+        buildArkoudaUdpServer(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag,distro=distro,tag=tag)
+    else:
+        buildChapelUdp(dockerRepo=dockerRepo,chapelVersion=chapelVersion,file=file,dockerTag=dockerTag)
+
+def getDistroName(distro: str, tag: Optional[str]) -> str:
+    if tag:
+        return tag
+    else:
+        return distro
+
+def generateChplSmpVersion(chapelVersion: str) -> str:
+    return f'chapel/chapel-gasnet-smp:{chapelVersion}'
+
+def generateChplUdpVersion(chapelVersion: str) -> str:
+    return f'bearsrus/chapel-gasnet-udp:{chapelVersion}'
+
+def generateArkoudaDownloadUrl(tag: Optional[str], branch: Optional[str]) -> str:
+    '''
+    Generates the Arkouda download URL based upon whether the desired Arkouda version
+    is either a tag or a branch.
+
+    :param Optional[str] tag: name of Arkouda tag, if applicable
+    :param Optional[str]: name of Arkouda branch, if applicable
+
+    :return: Arkouda download URL
+    :rtype: str
+    '''
+    if tag:
+        return f'https://github.com/Bears-R-Us/arkouda/archive/refs/tags/{tag}.zip'
+    elif branch:
+        return f'https://github.com/Bears-R-Us/arkouda/archive/refs/heads/{branch}.zip'
+    else:
+        raise ValueError('either the tag or branch must be not None')
+
+def generateBuildTag(dockerRepo: str, file: str, tag: Optional[str], distro: Optional[str]) -> str:
+    '''
+    Generates a docker build tag corresponding to the repo, file, tag, an distro
+
+    :param str dockerRepo: dockerhub repo the image will be published to
+    :param str file: Dockerfile name
+    :param Optional[str] tag: Arkouda tag name, if applicable
+    :param Optional[str] distro: Arkouda distro (branch name), if applicable
+
+    :return: docker build tag 
+    :rtype: str
+    '''
+    if tag:
+        return f'{dockerRepo}/{file}:{tag}'
+    else:
+        return f'{dockerRepo}/{file}:{distro}'   
```

Comment:
you could also do a ternary here if u decide u like the style

Reply:
updated here also, please check it out